mitodl/open-discussions
# Support generating user lists from playlists

#### Pre-Flight checklist

  - [x] Tag @Ferdi or @pdpinch 

#### What are the relevant tickets?
Closes #2367 

If you could also review https://github.com/mitodl/open-video-data/pull/6 as part of this since it's used in the testing anyway, it'd be much appreciated.

#### What's this PR do?
Adds loader code to generate user lists from playlists

#### How should this be manually tested?
- Set `OPEN_VIDEO_USER_LIST_OWNER` to a user that exists
- Set `OPEN_VIDEO_DATA_BRANCH=nl/uploads-no-user-list`
- Run `./manage.py backpopulate_youtube_data fetch -c UCTBMWu8yshnAmpzR3MoJFtw`
- Verify no user list was created
- Remove `OPEN_VIDEO_DATA_BRANCH` settings (this switches you back to `master` which has no `create_user_list` value set and will default to `True`
- Run `./manage.py backpopulate_youtube_data fetch -c UCTBMWu8yshnAmpzR3MoJFtw`
- Verify a user list was created, it contains the items from the playlist, and shows up in search

#### Screenshots (if appropriate)

Videos in a user list generated from a playlist:
![Screen Shot 2019-11-21 at 13 52 34](https://user-images.githubusercontent.com/28598/69367399-37c3d580-0c66-11ea-8300-d4b2f5356b87.png)
![Screenshot_2019-11-21 MIT Open Learning](https://user-images.githubusercontent.com/28598/69367400-37c3d580-0c66-11ea-95e2-55488b4ba14d.png)
